akka-cluster-sharding/src/main/scala/akka/cluster/sharding/ShardCoordinator.scala

```diff
@@ -889,6 +894,7 @@ abstract class ShardCoordinator(
 
   def regionTerminated(ref: ActorRef): Unit =
     if (state.regions.contains(ref)) {
+      rebalanceWorkers.foreach(_ ! RebalanceWorker.ShardRegionTerminated(ref))
```

Comment:
could there be a case where the state.regions doesn't contain the region any more, but the RebalanceWorker still has that ref. Might be safe to always send the ShardRegionTerminated to the RebalanceWorker?

Reply:
Can't hurt as RebalanceWorker doesn't validate it is a region it is waiting on